src/main/java/io/temporal/samples/hello/HelloSideEffect.java

```diff
@@ -0,0 +1,256 @@
+  }
+
+  /**
+   * Define the Activity Interface. Activities are building blocks of any temporal workflow and
+   * contain any business logic that could perform long running computation, network calls, etc.
+   *
+   * <p>Annotating activity methods with @ActivityMethod is optional
+   *
+   * @see io.temporal.activity.ActivityInterface
+   * @see io.temporal.activity.ActivityMethod
+   */
+  @ActivityInterface
+  public interface SideEffectActivities {
+
+    // Define your activity method which can be called during workflow execution
+    String greet(String greeting);
+  }
+
+  // Define the workflow implementation which implements our execute workflow method.
+  public static class SideEffectWorkflowImpl implements SideEffectWorkflow {
+
+    /**
+     * Define the SideEffectActivities stub. Activity stubs are proxies for activity invocations
+     * that are executed outside of the workflow thread on the activity worker, that can be on a
+     * different host. Temporal is going to dispatch the activity results back to the workflow and
+     * unblock the stub as soon as activity is completed on the activity worker.
+     *
+     * <p>Let's take a look at each {@link ActivityOptions} defined: The "setStartToCloseTimeout"
+     * option sets maximum time of a single Activity execution attempt. For this example it is set
+     * to 2 seconds.
+     */
+    private final SideEffectActivities activities =
+        Workflow.newActivityStub(
+            SideEffectActivities.class,
+            ActivityOptions.newBuilder().setStartToCloseTimeout(Duration.ofSeconds(2)).build());
+
+    int randomInt, sideEffectsRandomInt, unsafeRandomInt;
+    String randomUUID, sideEffectsRandomUUID;
+
+    @Override
+    public String execute() {
+
+      // Replay-safe way to create random number using Workflow.newRandom
+      randomInt = Workflow.newRandom().nextInt();
+
+      // Replay-safe way to create random uuid
+      randomUUID = Workflow.randomUUID().toString();
+
+      /*
+       * Random number using side effects. Note that this value is recorded in workflow history.
+       * On replay the same value is returned so determinism is guaranteed.
+       */
+      sideEffectsRandomInt =
+          Workflow.sideEffect(
+              int.class,
+              () -> {
+                Random random = new SecureRandom();
+                return random.nextInt();
+              });
+
+      /*
+       * Random ID using side effects. Note that this value is recorded in workflow history.
+       * On replay the same value is returned so determinism is guaranteed.
+       */
+      sideEffectsRandomUUID = Workflow.sideEffect(String.class, UUID.randomUUID()::toString);
+
+      /**
+       * Unsafe (not-deterministic way). We can query this after workflow execution to show that
+       * it's value changes.
+       */
+      unsafeRandomInt = new Random().nextInt();
+
+      /** Execute activity (sync) */
+      return activities.greet("World!");
```

Comment:
I would do something like:
```
String result = "Bye World!";
if ((randomUUID.version() + randomInt + sideEffectsRandomInt) % 2 == 0) {
   result = activities.greet("World");
 }
 return result;
```
Then if any of the calls is replaced with non safe random the queries are going to fail with non determinism.

Reply:
thanks!